prow/cmd/pipeline/main.go

```diff
@@ -0,0 +1,192 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	prowjobset "k8s.io/test-infra/prow/client/clientset/versioned"
+	prowjobinfo "k8s.io/test-infra/prow/client/informers/externalversions"
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/kube"
+	"k8s.io/test-infra/prow/logrusutil"
+	"k8s.io/test-infra/prow/pjutil"
+
+	pipelineset "github.com/tektoncd/pipeline/pkg/client/clientset/versioned"
+	pipelineinfo "github.com/tektoncd/pipeline/pkg/client/informers/externalversions"
+	pipelineinfov1alpha1 "github.com/tektoncd/pipeline/pkg/client/informers/externalversions/pipeline/v1alpha1"
+
+	"github.com/sirupsen/logrus"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp" // support gcp users in .kube/config
+)
+
+type options struct {
+	allContexts     bool
+	pipelineCluster string
+	config          string
+	kubeconfig      string
+	totURL          string
+}
+
+func parseOptions() options {
+	var o options
+	if err := o.parse(flag.CommandLine, os.Args[1:]); err != nil {
+		logrus.Fatalf("Invalid flags: %v", err)
+	}
+	return o
+}
+
+func (o *options) parse(flags *flag.FlagSet, args []string) error {
+	flags.BoolVar(&o.allContexts, "all-contexts", false, "Monitor all cluster contexts, not just default")
+	flags.StringVar(&o.totURL, "tot-url", "", "Tot URL")
+	flags.StringVar(&o.kubeconfig, "kubeconfig", "", "Path to kubeconfig. Only required if out of cluster")
+	flags.StringVar(&o.config, "config", "", "Path to prow config.yaml")
+	flags.StringVar(&o.pipelineCluster, "pipeline-cluster", "", "Path to file containing a YAML-marshalled kube.Cluster object. If empty, uses the local cluster.")
```

Comment:
This should still be called "build-cluster". The naming is bad, but this doesn't refer to knative builds, rather the cluster(s) where we execute workloads. We use the `--build-cluster` flag across Prow, not just for the Prow's knative build controller. e.g. https://github.com/kubernetes/test-infra/blob/0ffe02f69264e74009ef3856dd746f54961af2b1/prow/cmd/plank/main.go#L65

Reply:
done